cmd/participation/main.go

```diff
@@ -0,0 +1,169 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package main
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"path/filepath"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/hyperledger/fabric-protos-go/common"
+	"github.com/hyperledger/fabric/protoutil"
+	"github.com/pkg/errors"
+)
+
+// ChannelList represents multiple OSN channels.
+type channelList struct {
+	SystemChannel *channelInfo  `json:"systemChannel"`
+	Channels      []channelInfo `json:"channels"`
+}
+
+// ChannelInfo represents a channel.
+type channelInfo struct {
+	Name            string `json:"name,omitempty"`
+	URL             string `json:"url,omitempty"`
+	Status          string `json:"status,omitempty"`
+	ClusterRelation string `json:"clusterRelation,omitempty"`
+	Height          uint64 `json:"height,omitempty"`
+}
+
+func httpClient(tlsDir string) *http.Client {
+	clientCert, err := tls.LoadX509KeyPair(
+		filepath.Join(tlsDir, "server.crt"),
+		filepath.Join(tlsDir, "server.key"),
+	)
+	if err != nil {
+		panic(err)
+	}
+
+	clientCertPool := x509.NewCertPool()
+	caCert, err := ioutil.ReadFile(filepath.Join(tlsDir, "ca.crt"))
+	if err != nil {
+		panic(err)
+	}
+	clientCertPool.AppendCertsFromPEM(caCert)
+
+	return &http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{
+				Certificates: []tls.Certificate{clientCert},
+				RootCAs:      clientCertPool,
+			},
+		},
+	}
+}
+
+func getBodyBytes(client *http.Client, url string) ([]byte, error) {
+	resp, err := client.Get(url)
+	if err != nil {
+		return nil, err
+	}
+	if resp.StatusCode != http.StatusOK {
+		return nil, nil
+	}
```

Comment:
In case of a failed request that returns an error status code, the response contains an informative error message. It is better to print it as well.

Reply:
Refactored. Done.